### PR TITLE
Added OnDemand and Spot Price models addressing #131

### PR DESCRIPTION
**Pull request by mrcrgl**

### CODE REVIEW

**Comment by mrcrgl** [comment]
CLA signed

**Comment by mwielgus** [comment]
cc: @mumoshu @sethpollack

File: cluster-autoscaler/cloudprovider/aws/api/converter.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import "strconv"
+
+func stringRefToFloat64(p *string) (float64, error) {
```
Comment (mwielgus) [inline cluster-autoscaler/cloudprovider/aws/api/converter.go]: why pointer?

**Comment by mrcrgl** [comment]
I'll check that issues and re-open the pr

File: cluster-autoscaler/cloudprovider/aws/api/instance_info_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+func TestInstanceInfoService_DescribeInstanceInfo(t *testing.T) {
+	usEastOneURL, err := url.Parse(fmt.Sprintf(awsPricingAPIURLTemplate, "us-east-1"))
+	assert.NoError(t, err)
+
+	usWestOneURL, err := url.Parse(fmt.Sprintf(awsPricingAPIURLTemplate, "us-west-1"))
+	assert.NoError(t, err)
+
+	//usWestTwoURL, err := url.Parse(fmt.Sprintf(awsPricingAPIURLTemplate, "us-west-2"))
```
Comment (mwielgus) [inline cluster-autoscaler/cloudprovider/aws/api/instance_info_test.go]: remove or add more comment.

**Comment by mrcrgl** [comment]
@mwielgus sorry, I couldn't find you on slack. Would you please reach me out, my Slack name is mrcrgl.

File: cluster-autoscaler/cloudprovider/aws/api/instance_info.go [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+	"sync"
+	"time"
+)
+
+const instanceInfoCacheMaxAge = time.Hour * 6
+const awsPricingAPIURLTemplate = "https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/%s/index.json"
+
+// InstanceInfo holds AWS EC2 instance information
+type InstanceInfo struct {
+	InstanceType  string
```
Comment (mwielgus) [inline cluster-autoscaler/cloudprovider/aws/api/instance_info.go]: commens for fields please

File: cluster-autoscaler/cloudprovider/aws/api/instance_info.go [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+	Terms    terms              `json:"terms"`
+}
+
+type terms struct {
+	OnDemand map[string]productOffers `json:"OnDemand"`
+}
+
+type productOffers map[string]productOffer
+
+type productOffer struct {
```
Comment (mwielgus) [inline cluster-autoscaler/cloudprovider/aws/api/instance_info.go]: This seems to be some type of api definition. why it is not in AWS client library?

File: cluster-autoscaler/cloudprovider/aws/api/instance_info.go [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+}
+
+type product struct {
+	SKU        string            `json:"sku"`
+	Attributes productAttributes `json:"attributes"`
+}
+
+type productAttributes struct {
+	InstanceType    string `json:"instanceType"`
+	VCPU            string `json:"vcpu"`
```
Comment (mwielgus) [inline cluster-autoscaler/cloudprovider/aws/api/instance_info.go]: Either make the struct pubic of fields private. Here and elsewhere.

File: cluster-autoscaler/cloudprovider/aws/api/instance_spot_price_history.go [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+func spotPriceFilter(name string, values ...string) *ec2.Filter {
+	vs := stringToStringSliceRef(values...)
+
+	return &ec2.Filter{
+		Name:   &name,
+		Values: vs,
+	}
+}
+
+func convertSpotPriceItems(in ...*ec2.SpotPrice) SpotPriceItems {
```
Comment (mwielgus) [inline cluster-autoscaler/cloudprovider/aws/api/instance_spot_price_history.go]: What about error handling

**Comment by mwielgus** [comment]
This PR is HUGE. Please:

1) Consider splitting it into smaller, reviewable fragments.
2) Add a description how are you getting the prices and what is the exact logic.
3) Provide information on how you tested it in real world.

**Pull request closed without merging**